### ISSUE TITLE
Header template var "created" not correctly replaced when using pre-commit hook

### ISSUE THREAD

**Issue by trispo** [nicklockwood/SwiftFormat#640]
Hi,

I just found an issue when using swiftformat with the pre-commit hook from the Readme.

My `.swiftformat` file contains the following:

```
--header "\n Created by MyName on {created}.\n Copyright © {year} My Company. All rights reserved.\n"
```
When running `swiftformat` from the command line, the `created` and `year` templates are replaced correctly. But when using your provided pre-commit hook, the `created` template is not replaced whereas the `year` gets replaced correctly.

The result:
```
//
// Created by MyName on {created}.
// Copyright © 2020 My Company. All rights reserved.
//
```

I expected the `created` template var to be replaced, too.

**Comment by nicklockwood** [comment]
Thanks for reporting, I'll investigate.

**Comment by nicklockwood** [comment]
@trispo the issue is that the `git-format-staged` command is using stdin as the input, so it has no access to the actual file. I don't know if there's a simple way to rewrite this command to use the file path instead, but that's probably the way to solve it.

**Comment by nicklockwood** [comment]
@trispo I have a fix for this - I'll include it in the next release.

**Comment by trispo** [comment]
Sounds great. Thanks

**Comment by nicklockwood** [comment]
@trispo fixed in 0.44.15. Check README.md for updated pre-commit hook instructions.

**Comment by trispo** [comment]
@nicklockwood I tried the new version with the updated hook but unfortunately this doesn't work for me. A commit results in a committed version of my file where every line is deleted and a new (unstaged) file name stdout which contains the changes. Am I doing something wrong?

**Comment by nicklockwood** [comment]
Are you using version 0.44.15 of SwiftFormat? It only works with the latest version.

**Comment by trispo** [comment]
Yes
<img width="241" alt="image" src="https://user-images.githubusercontent.com/1045441/86331933-0f637700-bc4a-11ea-8fc8-ede3a0d6d651.png">

**Comment by nicklockwood** [comment]
Sorry about that, there was a bug. Can you try with 0.44.16? (you'll have to download the binary from this repo as homebrew will take a while to update).

**Comment by nicklockwood** [comment]
OK, it seems that there's a problem with the approach I was trying to take. You can't read from a file and also replace its contents with stdout in the same command - you just get a blank file. I'm not really sure why, but it's not a problem with SwiftFormat itself AFAICT.

I'll have to come up with a different solution. It won't be solved for this release I'm afraid.

**Comment by trispo** [comment]
OK, thanks for investigating further. You do an amazing work for the community.

**Comment by b-onc** [comment]
Hello @nicklockwood ,
Addition to the issue described in this ticket, is there a way to use `{file}` in header template while using `stdin` (because `git-format-staging`) as input?

**Comment by nicklockwood** [comment]
@b-onc no, it's not possible currently (for the same reason).

**Comment by nicklockwood** [comment]
@trispo @b-onc OK, this should be fixed (for real this time) in 0.44.17
